hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/validate/HazelcastSqlOperatorTable.java

```diff
@@ -402,6 +408,40 @@ private SqlCall rewriteCall(SqlCall call) {
             return call;
         }
 
+        @SuppressWarnings("CheckStyle")
+        private SqlCall expandViews(SqlCall call) {
+            if (call instanceof SqlSelect) {
+                SqlSelect selectCall = (SqlSelect) call;
+                ViewResolver viewResolver = validator.getViewResolver();
+
+                if (selectCall.getFrom() == null) {
+                    return call;
+                }
+
+                SqlNode from = selectCall.getFrom();
+                if (from instanceof SqlIdentifier) {
+                    SqlIdentifier fromClause = (SqlIdentifier) from;
+                    if (!ValidationUtil.isCatalogObjectNameValid(fromClause)) {
+                        // We are not throwing any exceptions here, delegating it to validation stage.
+                        return call;
+                    }
+                    String id = fromClause.names.get(fromClause.names.size() - 1);
+                    View resolvedView = viewResolver.resolve(id);
+                    if (resolvedView == null) {
+                        return call;
+                    } else {
+                        QueryParser parser = new QueryParser(validator);
+                        // Note: query was parsed & validated previously.
+                        // We assume, that no exception happens here during parsing.
+                        QueryParseResult parseResult = parser.parse(resolvedView.query());
```

Comment:
We must handle circular dependencies, like view1 reading from view2 and view2 from view1.